src/cli/reference_docs_generator_test.py

```diff
@@ -0,0 +1,141 @@
+from pathlib import Path
+
+from src.conftest import BarCommand, FooCommand
+from src.cli.cli_app import CLIApp
+from src.cli.command import Command
+from src.cli.reference_docs_generator import ReferenceDocsGenerator
+
+
+def test_generating_markdown_for_commands(
+    foo_command: FooCommand, bar_command: FooCommand
+):
+    docs_generator = ReferenceDocsGenerator(
+        CLIApp(commands=[foo_command, bar_command], root_args=[])
+    )
+
+    result = docs_generator.generate_cli_reference_markdown()
+    splitted_result = result.split("\n")
+
+    assert f"### `{bar_command.name}`" in splitted_result
+    assert bar_command.example is None
+    assert f"{bar_command.description}" in splitted_result
+
+    assert f"### `{foo_command.name}`" in splitted_result
+    assert f"{foo_command.example}" in splitted_result
+    assert f"{foo_command.description}" in splitted_result
+
+
+def test_generating_markdown_for_command_arguments(foo_command: FooCommand):
+    docs_generator = ReferenceDocsGenerator(
+        CLIApp(commands=[foo_command], root_args=[])
+    )
+
+    result = docs_generator.generate_cli_reference_markdown()
+    splitted_result = result.split("\n")
+
+    assert f"### `{foo_command.name}`" in splitted_result
```

Comment:
I wouldn't check the .md formatting, we might want to change it up in the future and testing for content only seems more intuitive and less error-prone